### PR TITLE
CompilerMSL pass builtin struct members into functions.

### PR DESCRIPTION
**Pull request by billhollings**
Add and use Compiler::get_non_pointer_type() convenience functions.

Fixes issue #610.

### CODE REVIEW

File: spirv_cross.cpp [hunk 1]
```diff
@@ -1063,6 +1063,34 @@ const SPIRType &Compiler::get_type_from_variable(uint32_t id) const
 	return get<SPIRType>(get<SPIRVariable>(id).basetype);
 }
 
+uint32_t Compiler::get_non_pointer_type_id(uint32_t type_id) const
```
Comment (HansKristian-ARM) [inline spirv_cross.cpp]: Good cleanup!

File: spirv_msl.cpp [hunk 2]
```diff
@@ -534,19 +534,44 @@ void CompilerMSL::extract_global_variables_from_function(uint32_t func_id, std::
...
+				type_id = get_non_pointer_type_id(type_id);
+				p_type = &get<SPIRType>(type_id);
 
-			next_id++;
+				uint32_t mbr_idx = 0;
+				for (auto &mbr_type_id : p_type->member_types)
+				{
+					BuiltIn builtin;
+					bool is_builtin = is_member_builtin(*p_type, mbr_idx, &builtin);
+					if (!is_builtin || has_active_builtin(builtin, var.storage))
```
Comment (HansKristian-ARM) [inline spirv_msl.cpp]: I don't understand this conditional. Shouldn't it be more like `is_builtin && has_active_builtin` instead?

File: reference/opt/shaders-msl/vert/set_builtin_in_func.vert [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+struct main0_out
+{
+    float4 gl_Position [[position]];
+};
+
+vertex main0_out main0()
+{
+    main0_out out = {};
+    out.gl_Position = float4(1.0);
```
Comment (HansKristian-ARM) [inline reference/opt/shaders-msl/vert/set_builtin_in_func.vert]: Could you add a write to gl_PointSize here so we can test that writing multiple builtins works?

File: spirv_msl.cpp (hunk 2 above)
Comment (billhollings) [reply spirv_msl.cpp]: Shoot! Lazy copy & paste from another test. Good catch!

File: reference/opt/shaders-msl/vert/set_builtin_in_func.vert (hunk 3 above)
Comment (billhollings) [reply reference/opt/shaders-msl/vert/set_builtin_in_func.vert]: Done.

**Comment by billhollings** [comment]
@HansKristian-ARM 

Is there a reason you didn't merge this PR? 

Did you want me to do it...or are you waiting for other dev activities?

**Comment by HansKristian-ARM** [comment]
I generally wait for Travis to complete before merging accepted PRs.

**Comment by billhollings** [comment]
Ah...okay...I figured Travis was done before your approval. Thanks for merging.

**Pull request merged**